### PR TITLE
Refactor/global state

### PR DESCRIPTION
**Pull request by taronaleksanian**

### CODE REVIEW

**Comment by andrew-bierman** [comment]
nice work with this, but i'm noticing by removing some dispatch calls some of our CRUD is breaking. I think a few of the places we had dispatch and custom hooks implemented is due to the custom hooks not working possibly. trying to gather list

File: packages/app/global-state/progressStore/progressStore.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import { atom } from 'jotai';
+import { produce } from 'immer';
+import { useReducerAtom } from 'jotai/utils';
+import * as Types from './types';
+
+import EventEmitter from 'events';
```
Comment (andrew-bierman) [inline packages/app/global-state/progressStore/progressStore.ts:6]: just curios why do we need these events and emitter logic in here ?

**Comment by andrew-bierman** [comment]
- [ ] double check favoritesStore conversion getting some errors on my side

File: packages/app/global-state/progressStore/progressStore.ts (hunk 1 above)
Comment (taronaleksanian) [reply packages/app/global-state/progressStore/progressStore.ts]: With jotai we can't dispatch actions for reducer atoms outside of the component, so I ended up creating this logic so we can do it, for the progress bar state. It seems we don't use the progress bar now, but it is there, so I think it's better to keep its support.

**Review by andrew-bierman — APPROVED** [review]
nice job with this

**Pull request merged**